pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -29,16 +28,6 @@ func (h *handler) onClusterChange(_ string, cluster *mgmtv3.Cluster) (*mgmtv3.Cl
 	if cluster == nil || cluster.DeletionTimestamp != nil {
 		return nil, nil
 	}
-	isK3s := cluster.Status.Driver == mgmtv3.ClusterDriverK3s
-	isRke2 := cluster.Status.Driver == mgmtv3.ClusterDriverRke2
-	// only applies to imported k3s/rke2 clusters
-	if !isK3s && !isRke2 {
-		return cluster, nil
-	}
-	// Don't allow nil configs to continue for given cluster type
-	if (isK3s && cluster.Spec.K3sConfig == nil) || (isRke2 && cluster.Spec.Rke2Config == nil) {
-		return cluster, nil
-	}
```

Comment:
Could we reinstate the check to verify whether it is an imported RKE2/K3s cluster? 
This is necessary to prevent attempts to uninstall the k3s-based-upgrader app on a non-imported rke2/k3s cluster.  - This is a new change since you raised this PR.

Reply:
Updated, checks exists it was just lower down due to resolving merge conflicts, moved it back up.